include/flatbuffers/verifier.h

```diff
@@ -25,22 +25,48 @@ namespace flatbuffers {
 // Helper class to verify the integrity of a FlatBuffer
 class Verifier FLATBUFFERS_FINAL_CLASS {
  public:
-  Verifier(const uint8_t *const buf, const size_t buf_len,
-           const uoffset_t _max_depth = 64,
-           const uoffset_t _max_tables = 1000000,
-           const bool _check_alignment = true)
+  struct Options {
+    // The maximum nesting of tables and vectors before we call it invalid.
+    uoffset_t max_depth = 64;
+    // The maximum number of tables we will verify before we call it invalid.
+    uoffset_t max_tables = 1000000;
+    // If true, verify all data is aligned.
+    bool check_alignment = true;
+    // If true, run verifier on nested flatbuffers
+    bool check_nested_flatbuffers = true;
+
+    // Manually implemented constructors for compilers that can't figure it out.
+    Options() = default;
+    Options(uoffset_t d, uoffset_t t, bool a, bool n)
+        : max_depth(d),
+          max_tables(t),
+          check_alignment(a),
+          check_nested_flatbuffers(n) {}
+  };
+
+
+  Verifier(const uint8_t *const buf, const size_t buf_len, const Options opts)
       : buf_(buf),
         size_(buf_len),
-        max_depth_(_max_depth),
-        max_tables_(_max_tables),
-        check_alignment_(_check_alignment),
+        opts_(opts),
         upper_bound_(0),
         depth_(0),
         num_tables_(0),
         flex_reuse_tracker_(nullptr) {
     FLATBUFFERS_ASSERT(size_ < FLATBUFFERS_MAX_BUFFER_SIZE);
   }
 
+  // Deprecated API, please construct with Verifier::Options.
+  Verifier(const uint8_t *const buf, const size_t buf_len,
+           const uoffset_t _max_depth = 64,
+           const uoffset_t _max_tables = 1000000,
+           const bool _check_alignment = true,
+           const bool _check_nested_flatbuffers = true)
+      : Verifier(buf, buf_len, {
```

Comment:
ah, i was just going to have it not call another constructor.

```
 Verifier(const uint8_t *const buf, const size_t buf_len,
           const uoffset_t _max_depth = 64,
           const uoffset_t _max_tables = 1000000,
           const bool _check_alignment = true) 
             : buf_(buf), etc...
           {
             opts_.max_depth = _max_depth;
             opts_.max_tables = _max_tables;
             etc...
             
           
           }
```

Reply:
I considered that but `opts_` and the replaced fields are const, and I chose to keep the const